server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -201,8 +201,16 @@ public enum MetadataVersion {
     // Add new fetch request version for KIP-951
     IBP_3_7_IV4(19, "3.7", "IV4", false),
 
+    // New version for the Kafka 3.8.0 release.
     // Add ELR related supports (KIP-966).
-    IBP_3_8_IV0(20, "3.8", "IV0", true);
+    IBP_3_8_IV0(20, "3.8", "IV0", false),
+
+    //
+    // NOTE: MetadataVersions after this point are unstable and may be changed.
+    // If users attempt to use an unstable MetadataVersion, they will get an error.
+    // Please move this comment when updating the LATEST_PRODUCTION constant.
+    //
+    IBP_3_9_IV0(21, "3.9", "IV0", false);
```

Comment:
1. The following comment needs to be moved to IBP_3_9_IV0.

`// Add ELR related supports (KIP-966).`

2. We need to change `isElrSupported()` to use IBP_3_9_IV0.
3. We need to bring all tests referencing IBP_3_9_IV1 for ELR in https://github.com/apache/kafka/pull/16347 here.
4. We need to bring in the changes in the following tests in https://github.com/apache/kafka/pull/16347 for testing latestTesting MV.
ClusterTestExtensionsTest.java
ZkMigrationIntegrationTest.scala
ApiVersionsRequestTest.scala
5. We need to add IBP_3_9_IV0 to all tests in MetadataVersionTest.java where 4.0-IV0 is used in trunk.

Reply:
Thanks @junrao
Would we need to add IBP_3_9_IV1 as well? Or just use the tests for IBP_3_9_IV1 using the IBP_3_9_IV0 enum value?

Reply:
I think I brought all necessary changes now